src/components/DomEvents.js

```diff
@@ -156,6 +121,23 @@ function DomEvents() {
     }
   }, []);
 
+  const typeOptions = Object.values(buffer.current).reduce(
+    (acc, { event }) =>
+      acc.findIndex((evtIncluded) => evtIncluded.value === event.EventType) ===
+      -1
+        ? [...acc, { label: event.EventType, value: event.EventType }]
+        : acc,
+    [],
+  );
+
+  const nameOptions = Object.values(buffer.current).reduce(
+    (acc, { event }) =>
+      acc.findIndex((evtIncluded) => evtIncluded.value === event.name) === -1
+        ? [...acc, { label: event.name, value: event.name }]
+        : acc,
+    [],
+  );
+
```

Comment:
maybe this?

```
const eventValues = Object.values(current)
const getUniqueEventsByProperty = (property) => [...new Set(eventValues.map(({ event }) => event[property]))]
const getOptionsByProperty = (property) => getUniqueEventsByProperty(property).map((eventType) => ({ label: eventType, value: eventType }))

const typeOptions = getOptionsByProperty('EventType')
const nameOptions = getOptionsByProperty('name')

Reply:
Hi @flaviohenriquecbc!

Thanks for taking the time to review this PR. When suggesting alternative code styles or solutions, I would appreciate some comment about why you prefer that style or think it's a better solution.

By doing so, we create a healthy environment for discussions. It enables the author to learn from it. Or it gives the author space to explain why they implemented it in the current state. It opens the discussion about the "reasons" instead of the "looks".

Again, thanks! I appreciate your review. And please keep doing so.

Reply:
On point your comment, @smeijer! I appreciate the healthy environment that you are creating. Ideas are there to be discussed and we are here also to learn and build things together. Thanks for the feedback.

@delca85 tks for your PR and appreciate your effort. About my previous comment, I think your code is quite good actually, I just brought this to the discussion, because while I was reading this specific excerpt I had to iterate over it to understand how exactly this was working. You used a clear logic and names for the variables, it made this effort easier. My thoughts were that we could merge both functions somehow because they are similar. I understand this is not critical, neither is wrong what you are suggesting. Since you spent time on this, I thought it would be interesting to challenge myself to understand your code and learn from it, and also try to contribute somehow to your work.

Anyway, good work! Keep moving on :)

Reply:
@flaviohenriquecbc thank you very much for your comment, I think working on this project is a very chance for me to learn and your advice confirms that!
I replaced my code with the one you suggested me that sounds more clear to me too.
Thanks!